src/auth/guards/role.guard.ts

```
@@ -0,0 +1,40 @@
+    canActivate(context: ExecutionContext): boolean {
+      // Get required roles from metadata
+      const requiredRoles = this.reflector.getAllAndOverride<UserRole[]>(ROLES_KEY, [
+        context.getHandler(),
+        context.getClass(),
+      ]);
+     
+  
+      // Get user from request
+      const request = context.switchToHttp().getRequest();
+      const user = request.user; // Ensure user is attached to request (e.g., via JWT auth)
+      console.log('User from request:', user);
```

ESLint

[error] Delete `··`

(prettier/prettier)